nixos/modules/system/boot/loader/external/external.nix

```diff
@@ -0,0 +1,38 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.boot.loader.external;
+in
+{
+  meta = {
+    maintainers = with maintainers; [ cole-h grahamc ];
+    # Don't edit the docbook xml directly, edit the md and generate it:
+    # `pandoc external.md -t docbook --top-level-division=chapter --extract-media=media -f markdown+smart > external.xml`
+    doc = ./external.xml;
+  };
+
+  options.boot.loader.external = {
```

Comment:
I actually don't like the name here, considering in-tree bootloaders will probably want to use this as well. Maybe just have `boot.loader.installHook` everywhere and either ignore it if it's null or find some way to hook the old builders into it?

Reply:
Why would they want to use it?

Reply:
Just for uniformity? Like, the way I see it, the final goal would be to just have all bootloader backends set this one option.

Reply:
Do you mean they may want to use bootspec, or the `.external` namespace?

Reply:
I mean that they'd want to use bootspec and just set the `installer` option to their own installer instead of something user provided.

Reply:
They'd be able to use bootspec and wouldn't need to use the "external" option. Every generation gets the bootspec file. This is defining the interface to how to specify an external bootloader. I think they can cohabitate via this option?

Reply:
I propose we move this to another PR to track this thing once we have a more polished bootspec in-tree.